src/functionalTest/groovy/com/autonomousapps/android/AllDependenciesSpec.groovy

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2024. Tony Robalik.
+// SPDX-License-Identifier: Apache-2.0
+package com.autonomousapps.android
+
+import com.autonomousapps.android.projects.DuplicateDependencyVersionsProject
+import org.gradle.util.GradleVersion
+
+import static com.autonomousapps.kit.truth.BuildResultSubject.buildResults
+import static com.autonomousapps.utils.Runner.build
+import static com.google.common.truth.Truth.assertAbout
+import static com.google.common.truth.Truth.assertThat
+
+final class AllDependenciesSpec extends AbstractAndroidSpec {
+
+  def "can generate a version catalog file with all dependencies (#gradleVersion AGP #agpVersion)"() {
+    given:
+    def project = new DuplicateDependencyVersionsProject(agpVersion as String)
```

Comment:
Using this existing project fixture `DuplicateDependencyVersionsProject` because it had what I needed, happy to rename it to be more generic or create a new one for this test.